modules/statistics/php/stats_demographic.class.inc

```diff
@@ -163,65 +167,80 @@ class Stats_Demographic extends \NDB_Form
         foreach (\Utility::getProjectList() as $key => $value) {
             $projects[$key] = $value;
         }
-        $currentProject = null;
-        if ($_REQUEST['DemographicProject'] ?? ''
-        ) {
-            $currentProject = htmlspecialchars($_REQUEST['DemographicProject']);
+        $demographicProject = $_REQUEST['DemographicProject'] ?? '';
+        $Param_Project      = '';
+
+        if (empty($demographicProject)) {
+            error_log('No demographic project specified in request');
```

Comment:
exception ?

Reply:
No. Currently, if `$demographicProject` is null, the part of the result query defined by it is ignored.

Reply:
see above

Reply:
In that case there should be no error logged, because it's the expected behaviour

Reply:
what about lines 194, 210, and 230?

Reply:
It applies to all the error_log calls. Either exception if its not expected or nothing.

You can put the content of the error log as a comment if it helps with readability